### PR TITLE
feat: create UI components pie chart

### PR DESCRIPTION
**Pull request by yuki-92c**
## Overview

- Add pie chart component in the components/ui folder based on the design on Figma.

## Changes

- Style the design

## Review points

- The styling matches the Figma design

## Screenshots or videos

- figma design
image

- pie chart component
image

## Notes

- example code

```ts
import { Progress } from '../components/ui/piechart';

export default function Home() {
  return (
    <main className="flex min-h-screen flex-col items-center justify-between p-24">
      <div className="h-full w-full">
          <Progress progressPercent={90} />
      </div>
    </main>
  );
}

```

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/piechart.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          to {
+            transform: rotate(${progressPercent * 3.6}deg)
+          }
+        }
+        `}
+      </style>
+      <div className="relative">
+        <svg
+          viewBox={`0 0 ${size} ${size}`}
+          style={{ transform: 'rotate(-90deg)' }}
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you modify this code with TailwindCSS class?
`className="transform rotate-90"`

File: src/components/ui/piechart.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        >
+          <circle
+            r={radius}
+            cx={size / 2}
+            cy={size / 2}
+            stroke="#CBD5E1"
+            strokeWidth="15"
+            fill="transparent"
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you modify this code with TailwindCSS class and our color class?
`primary-light`

File: src/components/ui/piechart.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            strokeDasharray={circumference}
+          />
+          <circle
+            r={radius}
+            cx={size / 2}
+            cy={size / 2}
+            stroke="#06B6D4"
+            strokeWidth="15"
+            fill="transparent"
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you modify this code with TailwindCSS class and our color class?
`slate-300`

File: src/components/ui/piechart.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            strokeDasharray={circumference}
+            style={{
+              animation: 'circleStroke 2s ease forwards',
+            }}
+          />
+        </svg>
+        <div
+          className="absolute top-0 left-1/2 "
+          style={{
+            translate: '-50% -50%',
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you modify this code with TailwindCSS class?
`transform -translate-x-1/2 -translate-y-1/2`

File: src/components/ui/piechart.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            transformOrigin: `center ${256 + 126 / 2}px`,
+            animation: 'rotationObject 3s ease forwards',
+          }}
+        ></div>
+        <div
+          className="absolute top-1/2 left-1/2 text-5xl text-black whitespace-nowrap"
+          style={{
+            translate: '-50% -50%',
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you modify this code with TailwindCSS class?
`transform -translate-x-1/2 -translate-y-1/2`

File: src/components/ui/piechart.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            style={{
+              animation: 'circleStroke 2s ease forwards',
+            }}
+          />
+        </svg>
+        <div
+          className="absolute top-0 left-1/2 "
+          style={{
+            translate: '-50% -50%',
+            transformOrigin: `center ${256 + 126 / 2}px`,
```
Comment (Akikaze1119) [inline src/components/ui/piechart.tsx]: Could you tell me the reason why you use `${256 + 126 / 2}px`, not `319px`?

**Review by Akikaze1119 — CHANGES_REQUESTED** [review]
Thank you for your work.
I'd like you check my comments.

File: src/components/ui/piechart.tsx (hunk 2 above)
Comment (yuki-92c) [reply src/components/ui/piechart.tsx]: Thanks for your review. I'd like to discuss about the stroke width later

File: src/components/ui/piechart.tsx (hunk 3 above)
Comment (yuki-92c) [reply src/components/ui/piechart.tsx]: same as the one above

**Review by Akikaze1119 — APPROVED** [review]
Thank you for fixing!
LGTM :)

**Pull request merged**